subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/NonHierarchicalFileWatcherUpdater.java

```diff
@@ -72,6 +72,7 @@ public void changed(Collection<CompleteFileSystemLocationSnapshot> removedSnapsh
 
     @Override
     public void buildFinished() {
+        LOGGER.warn("Watching {} directories to track changes", watchedRoots.entrySet().size());
```

Comment:
No need for `entrySet()`.

Reply:
`entrySet()` is necessary, since `watchedRoots` is a `Multiset`.

Reply:
Right, that makes sense.